site/e2e/helpers.ts

```diff
@@ -709,6 +713,30 @@ export const updateTemplate = async (
   await uploaded.wait();
 };
 
+export const updateTemplateSettings = async (
+  page: Page,
+  templateName: string,
+  templateSettingValues: Pick<
+    UpdateTemplateMeta,
+    "name" | "display_name" | "description"
+  >,
+) => {
+  await page.goto(`/templates/${templateName}/settings`, {
+    waitUntil: "domcontentloaded",
+  });
+  await expect(page).toHaveURL(`/templates/${templateName}/settings`);
+
+  for (const [key, value] of Object.entries(templateSettingValues)) {
```

Comment:
There's nothing wrong with the code right now, but there is a risk in the future from runtime/compile-time mismatches:

This will not throw a type error:
```ts
const myObj = {
  name: "whoops",
  obviouslyWrongValue: true
};

updateTemplateSettings(page, templateName, myObj);
```
You only get the error if you pass the object in as a literal, but there's not a great way to enforce people call the function that way:
```ts
updateTemplateSettings(page, templateName, {
  name: "whoops",
  
  // Error: Object literal may only specify known properties, and 'obviouslyWrongValue' does not exist in...
  obviouslyWrongValue: true
});
```
And because the loop is iterating over everything (regardless of what the type info says), you will get both properties logged, not just `name`

Just to be on the safe side, and because this is a general test helper, I'm wondering if there should be a runtime check on the key to make sure that you're only processing name/display_name/description

Reply:
Edited the message because I didn't like the way my tone sounded. Sorry about that